### PR TITLE
fix: Size for `SpriteComponent.fromImage` should be nullable

### PR DESCRIPTION
**Pull request by fa-fifi**
# Description

I received an assertion error after I set `autoResize` to true when I am using `SpriteComponent.fromImage` component. It's kinda weird for me that the message tells me **"If size is set, autoResize should be false or size should be null when autoResize is true.''**, but the thing is I never set the size for that image yet. Then, I check the code and I found out that the size will never be null anyway, so the `autoresize` should always be false.

In this PR, I just remove all the fallback value for `size`, so it would not give an assertion error when `size` is not being set yet while the `autoresize` is true. 

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [x] No, this PR is not a breaking change.

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: packages/flame/lib/src/components/sprite_component.dart [hunk 1]
```diff
@@ -72,7 +71,6 @@ class SpriteComponent extends PositionComponent
             srcPosition: srcPosition,
             srcSize: srcSize,
           ),
-          autoResize: autoResize,
           paint: paint,
           position: position,
           size: size ?? srcSize ?? image.size,
```
[the remark this answers is not in the document]
Comment (spydon) [reply packages/flame/lib/src/components/sprite_component.dart:76]: What's the reason for that? Then it'll have a different behavior than the other constructor right? :thinking:

[the remark this answers is not in the document]
Comment (ufrshubham) [reply packages/flame/lib/src/components/sprite_component.dart:76]: Not really. `SpriteComponent.fromImage` internally invokes the same constructor. 

https://github.com/flame-engine/flame/blob/e434bafb15fc486c51b43aaa9d9190b8b7e783cb/packages/flame/lib/src/components/sprite_component.dart#L69

[the remark this answers is not in the document]
Comment (fa-fifi) [reply packages/flame/lib/src/components/sprite_component.dart:76]: So... should I commit changes based on @ufrshubham's suggestion?

[the remark this answers is not in the document]
Comment (spydon) [reply packages/flame/lib/src/components/sprite_component.dart:76]: Yeah, please do

[the remark this answers is not in the document]
Comment (ufrshubham) [reply packages/flame/lib/src/components/sprite_component.dart:76]: Yup, commit that change and then `autoResize` can be left as it was earlier.

**Review by spydon — APPROVED** [review]

**Review by ufrshubham — CHANGES_REQUESTED** [review]
@fa-fifi can you update the title and description of the PR to be inline with what it actually does. Also, some tests for `SpriteComponent.fromImage` would be good.

**Review by ufrshubham — APPROVED** [review]

**Pull request merged**